core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -1310,6 +1309,24 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties with Logging {
       assert(Utils.trimExceptCRLF(s"b${s}b") === s"b${s}b")
     }
   }
+
+  test("deleting an already opened file does not interrupt the reading process") {
```

Comment:
Not sure why you're adding this test? That's standard posix behavior. (On Windows the delete might fail but the read should still work.)

If you're adding this as a proof that this actually works so that the feature you're adding doesn't break, then that should be explained. But in that case I'd just put a big comment in the part of the code that might be affected by this and skip this test.

Reply:
The test is removed. 

I think these comments should be enough:
https://github.com/apache/spark/blob/49f950893aa2d41bfddb7218613f070eefb9ba9e/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L848

https://github.com/apache/spark/blob/49f950893aa2d41bfddb7218613f070eefb9ba9e/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L853-L854